### PR TITLE
Scan for i2c sensors in environmental telemetry if enabled

### PR DESCRIPTION
**Pull request by thebentern**
1. Added bosch sensors to i2c scan (with sensor type array map for now)
2. Moved i2c sensors out of the case logic for sensor types set via the cli. (Seems like a sensible default to read em if you got em)
4. Established an hierarchy by ordering sensor measurements by quality (highest quality at the end so that the last temperature reading wins for instance). This allows a primitive form of gathering data from multiple sensors until we can come up with a better mechanism for sending discrete sensor telemetry messages.

### CODE REVIEW

**Comment by thebentern** [comment]
I'll take a look at the linter warning tomorrow.

File: src/debug/i2cScan.h [hunk 1]
```diff
@@ -104,6 +105,15 @@ void scanI2Cdevice(void)
                 DEBUG_MSG("axp192 PMU found\n");
             }
 #endif
+        if (addr == BME_ADDR || addr == BME_ADDR_ALTERNATE) {
+            // Assume BME680 first until we find out otherwise in environmental telemetry init
```
Comment (caveman99) [inline src/debug/i2cScan.h:109]: Is there some sub register you can use here to tell 280 and 680 apart? Like when we scan for cardkb or rak-kb. If not, we can leave it like it is with the detect later.

File: src/modules/Telemetry/EnvironmentTelemetry.cpp [hunk 2]
```diff
@@ -73,32 +75,31 @@ int32_t EnvironmentTelemetryModule::runOnce()
...
-                DEBUG_MSG("Environment Telemetry: Invalid sensor type selected; Disabling module");
-                return (INT32_MAX);
+                case TelemetrySensorType_DHT11:
+                case TelemetrySensorType_DHT12:
+                case TelemetrySensorType_DHT21:
+                case TelemetrySensorType_DHT22:
+                    result = dhtSensor.runOnce();
+                case TelemetrySensorType_DS18B20:
+                    result = dallasSensor.runOnce();
+                default:
```
Comment (caveman99) [inline src/modules/Telemetry/EnvironmentTelemetry.cpp:87]: Up till here you will always return dallasSensor. The old code "return"ed, e.g. jumped out of the switch, this one will just fall through with a new assignment. add a 'break' after dhtSensor.runOnce(); and dallasSensor.runOnce();

File: suppressions.txt [hunk 3]
```diff
@@ -29,6 +29,7 @@ useInitializationList
 unreadVariable
 
 redundantInitialization
+redundantAssignInSwitch
```
Comment (caveman99) [inline suppressions.txt:32]: After the correction above for the switch statement, this can be removed again.

**Review by caveman99 — CHANGES_REQUESTED** [review]
Looking ace, with a few remarks...

File: src/modules/Telemetry/EnvironmentTelemetry.cpp (hunk 2 above)
Comment (thebentern) [reply src/modules/Telemetry/EnvironmentTelemetry.cpp:87]: Oh good catch. I got code blind and completely missed that when I was gutting that init method

File: src/debug/i2cScan.h (hunk 1 above)
Comment (thebentern) [reply src/debug/i2cScan.h:109]: My google-fu didn't come up with anything. I'll skim the Bosch docs really quick to see if I can dig up something there

**Pull request merged**